libs/threading_base/src/thread_helpers.cpp

```diff
@@ -204,6 +205,33 @@ namespace hpx { namespace threads {
         return get_thread_id_data(id)->set_thread_data(data);
     }
 
+#ifdef HPX_HAVE_LIBCDS
+    cds_type get_libcds_data(thread_id_type const& id, error_code& ec)
+    {
+        if (HPX_UNLIKELY(!id))
+        {
+            HPX_THROWS_IF(ec, null_thread_id, "hpx::threads::get_libcds_data",
+                "null thread id encountered");
+            return cds_type{{0, 0, 0}};
```

Comment:
@hkaiser I agreed with passing by reference and return by reference, but what should I do for this return which returns a copy of cds data if `HPX_UNLIKELY(!id) == true`? Any suggestion?

Reply:
What about:
```
static cds_type data({{0, 0, 0}});
return data;
```
?